bertopic/_bertopic.py

```diff
@@ -4527,9 +4530,13 @@ def _sort_mappings_by_frequency(self, documents: pd.DataFrame) -> pd.DataFrame:
             documents: Updated dataframe with documents and the mapped
                        and re-ordered topic ids
         """
-        self._update_topic_size(documents)
+        # No need to sort if it's the first pass of zero-shot topic modeling
+        nr_zeroshot = len(self._topic_id_to_zeroshot_topic_idx)
+        if self._is_zeroshot and not self.nr_topics and nr_zeroshot > 0:
+            return documents
```

Comment:
I thought the plan was to update `self._topic_id_to_zeroshot_topic_idx` in `._sort_mappings_by_frequency` if zeroshot, and then *not* use `self.topic_mapper_.get_mappings()` in `.topic_labels_`?

Reply:
Ah right, I wasn't familiar enough with `self._topic_id_to_zeroshot_topic_idx` so wasn't sure the best way to change it. This seemed faster and is working now but if you have suggestions on how to change it, feel free to share.